### PR TITLE
refactor: revamp snap `index.ts` error handle

### PR DESCRIPTION
**Pull request by stanleyyconsensys**
This PR is to refactor the behaviour on how `index.ts` handle the error

The actual error message should not mask if it is expected to throw , e.g SnapError
Else it should mask with an general error message
Rather than depends on the log level to show / hide the actual error

The PR changes include:

- change error handle
  - from:
    - throw general error message if log level === off
    - throw actual error message if log level !== off
  - to:
    - throw general error message if the error is not a Snap Error
    - throw error message if the error is a Snap Error (including validation error)

- remove duplicate error handle layer from `RpcController`

### Requirement:
Requires this PR to be merged first:

- [x] #342

### CODE REVIEW

File: packages/starknet-snap/src/index.test.ts [hunk 1]
```diff
@@ -50,15 +51,30 @@ describe('onRpcRequest', () => {
...
+    const { createAccountSpy } = createMockSpy();
+    createAccountSpy.mockRejectedValue(new SnapError('error'));
 
-    await expect(onRpcRequest(createMockRequest())).rejects.toThrow(
-      'Unable to execute the rpc request',
-    );
+    await expect(onRpcRequest(createMockRequest())).rejects.toThrow(SnapError);
+  });
+
+  it('throws `SnapError` if the error is not an instance of SnapError', async () => {
```
Comment (khanti42) [inline packages/starknet-snap/src/index.test.ts:73]: This wording is a bit misleading. Does this mean we always throw SnapError ?

Comment (khanti42) [reply packages/starknet-snap/src/index.test.ts:73]: Maybe we should link it with the log level ?

File: packages/starknet-snap/src/index.ts [hunk 2]
```diff
@@ -293,19 +295,18 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
...
-      ) as unknown as Error;
-    } else {
-      throw new SnapError(error.message) as unknown as Error;
+    let snapError = error;
+
+    if (!isSnapRpcError(error)) {
+      snapError = new SnapError('Unable to execute the rpc request');
     }
+    logger.error(
+      `onRpcRequest error: ${JSON.stringify(snapError.toJSON(), null, 2)}`,
```
Comment (khanti42) [inline packages/starknet-snap/src/index.ts:307]: Shall we add a test for this logger.error to ensure that it's shown when log level is error ?

File: packages/starknet-snap/src/index.test.ts (hunk 1 above)
Comment (stanleyyconsensys) [reply packages/starknet-snap/src/index.test.ts:73]: yes we always throw SnapError , it has nothing to do with log level

File: packages/starknet-snap/src/index.ts (hunk 2 above)
Comment (stanleyyconsensys) [reply packages/starknet-snap/src/index.ts:307]: no need, as the logger test is already tested the feature

**Review by khanti42 — APPROVED** [review]
LGTM

**Pull request merged**